dev/release/versions.env

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# The release as a whole has a date-based identifier.  This is used to
+# identify tags, branches, and so on.
+RELEASE="2024.05.06"
```

Comment:
Hmm. It may be better that we omit preceding `0`: `2024.5.6`
This is not a semantic versioning but semver https://pypi.org/project/semver/ can't parse `2024.05.06` format: 

```console
>>> import semver
>>> semver.VersionInfo.parse('1.02.03')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/semver.py", line 656, in parse
    raise ValueError("%s is not valid SemVer string" % version)
ValueError: 1.02.03 is not valid SemVer string
```

(Sorry. The above transcript uses old semver, 2.10.2.)

Reply:
Ok! I'll take another look this afternoon

What do you think of just using a counter in that case then? (2024.1, 2024.2, 2024.3)

Reply:
> What do you think of just using a counter in that case then? (2024.1, 2024.2, 2024.3)

It will work too!

It may be confused but we may be able to omit the `2024.` part: `1`, `2`, `3`

We need to choose the next identifier when we publish a new release. If we use date-based identifier, it may be difficult. For example, we want to publish a next release in this year but it may be slipped to the next year.

(Should we choose the next identifier when we publish the next release?)

Reply:
Yeah, I think we can always adjust the identifier at publication time so it will be OK if we have to change it. We can always make the scripts a bit more flexible in that case too so we can update the version at any time.

Reply:
But yeah, in that case I think having the identifier just be 1, 2, 3, makes more sense.

Reply:
I suppose there might be a little confusion for our first release. But after that we will have release = 2, version number = 1.1.0/0.13.0 which should be OK.

Reply:
Ok, I went with "12.0.0" since this will be our 12th release.

Reply:
Ah...the version in the docs will look a little funny. I suppose that's OK for now. We can always label the docs with a date or something later.

Reply:
Can we use "12" instead of "12.0.0" because we will always use ".0.0" for the part.

Reply:
Sure, done